### DISCUSSION TITLE
How do I globally disable prerendering?

### DISCUSSION THREAD

**Discussion by steve-taylor** [sveltejs/kit#10771] [Q&A]
I want to completely disable pre-rendering. I only want server-side-rendered pages to be dynamically rendered. I don't want the build process to execute any of my code at build time. How do I globally configure SvelteKit to not pre-rendering in one place as opposed to having to remember to disable it on every single page as I keep adding more pages?

**Comment by teemingc** [comment]
Prerender is disabled by default. However:

> you can set export const prerender = true in your root +layout.js or +layout.server.js and prerender everything except pages that are explicitly marked as not prerenderable:

```js
// src/routes/+layout.js
export const prerender = false; // this disables prerender for all pages affected by the layout
```

https://kit.svelte.dev/docs/page-options#prerender

**Reply by steve-taylor** [reply]
> Prerender is disabled by default.

If that's the case, why does `vite build` execute my code, specifically hooks.server.ts? What reason does it have to execute that code if not to prerender pages?

**Reply by teemingc** [reply]
It could be that a fallback page is being generated. Are you using adapter-cloudflare? See https://github.com/sveltejs/kit/issues/9386

Otherwise, you’d have to provide a minimal reproduction so we can diagnose the root cause

**Reply by steve-taylor** [reply]
I'm using the SST adapter, i.e.
```ts
import adapter from 'svelte-kit-sst'
```

**Reply by teemingc** [reply]
The adapter isn't generating a fallback page. https://github.com/sst/sst/blob/master/packages/svelte-kit-sst/src/index.ts

There might be a setting in one of your layouts where prerender is set to true.

**Reply by steve-taylor** [reply]
None of my layouts or pages enable prerendering. The `prerender` token exists nowhere inside my project.

**Reply by teemingc** [reply]
I'm not sure what else it could be at this point. You'd have to provide a reproduction of this behaviour to debug it.

**Reply by steve-taylor** [reply]
To be perfectly honest, that's not in the cards right at this moment. I'll use the `building` flag for now. Thanks for your help.

**Reply by AlanBreck** [reply]
> If that's the case, why does vite build execute my code, specifically hooks.server.ts? What reason does it have to execute that code if not to prerender pages?

It runs all of your server code for analysis at build time, but not necessarily to prerender your pages. https://kit.svelte.dev/docs/building-your-app#during-the-build

> We need to set prerender to false in src/routes/+page.ts since we want to generate the presigned URL on page load.
>
> https://docs.sst.dev/start/svelte

Stab in the dark, but from looking at the doc page referenced above, I'm wondering if you may need to explicitly set `export const prerender = false`.
